File: Labs/Lab5/TreeNode.java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }

    public static boolean isValidBST(TreeNode root) { //lab5
        if (root == null) {
            return true;
        }
        if (root.left != null) {
            if (root.val < root.left.val) {
                return false;
            }
        }
        if (root.right != null) {
            if (root.val > root.right.val) {
                return false;
            }
        }
        return (isValidBST(root.left) && isValidBST(root.right));
    }
}

File: Labs/Lab5/TreeNode.py
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


    def isValidBST(self, root): #lab5
        if root is None: return True
        if root.left is not None:
            if root.val < root.left.val:return False
        if root.right is not None:
            if root.val > root.right.val:return False
        return self.isValidBST(root.left) and self.isValidBST(root.right)
